src/OSPSuite.Core/Domain/Builder/SimulationBuilder.cs

```diff
@@ -88,24 +87,97 @@ public IReadOnlyList<string> AllPresentEndogenousStationaryMoleculeNames() =>
 
       private void performMerge()
       {
+         cacheBuilders(x => x.Reactions, _reactions);
+         cacheBuilders(x => x.Molecules, _molecules);
          cacheMoleculeDependentBuilders(x => x.PassiveTransports, _passiveTransports);
-         _reactions.AddRange(allBuilder(x => x.Reactions));
          cacheMoleculeDependentBuilders(x => x.Observers, _observers);
-         _molecules.AddRange(allBuilder(x => x.Molecules));
-         _parameterValues.AddRange(allStartValueBuilder(x => x.SelectedParameterValues));
+         cacheParameterValueBuilders(x => x.SelectedParameterValues, _parameterValues);
+         cacheInitialConditions();
+         cacheEntities();
+      }
+
+      private void cacheEntities()
+      {
+         cacheContainers(_reactions);
```

Comment:
While I could cache all reactions etc, I am only caching the one that are going to be used after merging. Not sure if this is very useful or not. I am not doing it for the spatial structure.  I might revert this to simply cache all of them even if not used as this will simplify the code. Thoughts?

Reply:
Yeah forget it. I simplified it